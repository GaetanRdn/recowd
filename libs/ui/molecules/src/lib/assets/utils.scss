@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
