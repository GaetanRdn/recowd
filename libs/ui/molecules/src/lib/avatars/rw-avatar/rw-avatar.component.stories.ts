import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { RwAvatarComponent } from './rw-avatar.component';

export default {
  title: 'molecules/avatars/rw-avatar',
  decorators: [
    moduleMetadata({
      imports: [RwAvatarComponent],
    }),
  ],
} as Meta<RwAvatarComponent>;

const Template: Story<RwAvatarComponent> = () => ({
  template: `<rc-rw-avatar></rc-rw-avatar>`,
});

export const Default = Template.bind({});
