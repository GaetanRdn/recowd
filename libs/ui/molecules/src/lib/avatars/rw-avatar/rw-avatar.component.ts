import { ChangeDetectionStrategy, Component } from '@angular/core';
import { AvatarDirective } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-rw-avatars',
  standalone: true,
  imports: [AvatarDirective],
  template: `<img alt="avatar" rcAvatar src="assets/venom.png" />`,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class RwAvatarComponent {}
