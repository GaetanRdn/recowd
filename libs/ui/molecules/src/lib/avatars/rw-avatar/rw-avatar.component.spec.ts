import { TestBed, waitForAsync } from '@angular/core/testing';

import { RwAvatarComponent } from './rw-avatar.component';
import { TemplateLookup } from '@recowd/test/utils';
import { Component } from '@angular/core';

describe('RwAvatarComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [RwAvatarComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});

@Component({
  template: ` <rc-rw-avatars></rc-rw-avatars>`,
})
class HostComponent {}
