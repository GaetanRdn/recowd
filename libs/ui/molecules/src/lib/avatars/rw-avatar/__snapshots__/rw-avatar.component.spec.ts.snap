// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RwAvatarComponent should create 1`] = `
<rc-rw-avatar>
  <img
    alt="avatar"
    class="rc-avatar"
    rcavatar=""
    src="assets/venom.png"
  />
</rc-rw-avatar>
`;
